### PR TITLE
mSize in GetVolumes fix.

### PR DESCRIPTION
**Pull request by valentinbreiz**
Now mSize in GetVolumes works and won't return 0.

### CODE REVIEW

File: source/Cosmos.System2/FileSystem/FAT/Listing/FatDiretoryEntry.cs [hunk 1]
```diff
@@ -44,7 +44,7 @@ internal class FatDirectoryEntry : DirectoryEntry
             string aFullPath,
             string aName,
             uint aFirstCluster)
-            : base(aFileSystem, aParent, aFullPath, aName, 0, DirectoryEntryTypeEnum.Directory)
+            : base(aFileSystem, aParent, aFullPath, aName, aSize, DirectoryEntryTypeEnum.Directory)
```
Comment (jp2masa) [inline source/Cosmos.System2/FileSystem/FAT/Listing/FatDiretoryEntry.cs]: `aSize` doesn't exist here.

Comment (valentinbreiz) [inline source/Cosmos.System2/FileSystem/FAT/Listing/FatDiretoryEntry.cs]: Done.

File: source/Cosmos.System2/FileSystem/CosmosVFS.cs [hunk 2]
```diff
@@ -355,10 +355,12 @@ protected virtual void InitializeFileSystems()
             for (int i = 0; i < mPartitions.Count; i++)
             {
                 string xRootPath = string.Concat(i, VolumeSeparatorChar, DirectorySeparatorChar);
+                ulong Size = mPartitions[i].BlockCount * mPartitions[i].BlockSize / 1024 / 1024;
```
Comment (jp2masa) [inline source/Cosmos.System2/FileSystem/CosmosVFS.cs]: You can make the cast here:
```cs
ulong Size = (long)(mPartitions[i].BlockCount * mPartitions[i].BlockSize / 1024 / 1024);
```

**Comment by fanoI** [comment]
In reality I'm not sure if make a public property return an unsigned type is so much correct... it will not be anymore CLS compliant (so J#, Javascript.net and IronPython for example could not "consume" it).
In the end long (Int64) is really big and well if a partion is more bigger that 2^64 Byte you have problems in any case.

**Comment by jp2masa** [comment]
2^63 - 1, right?

**Comment by fanoI** [comment]
Yes that is the number: 9,223,372,036,854,775,807 that should be sufficient at least until we reach the hexabyte HDD drives :-)

**Comment by valentinbreiz** [comment]
So is it ready to merge ?

**Comment by jp2masa** [comment]
See my revision. `ulong` should be changed to `long`, I hadn't noticed that DirectoryEntry uses long for size.

**Comment by valentinbreiz** [comment]
I changed that.

**Comment by jp2masa** [comment]
You changed `long` to `ulong`, but you need to revert that change.

**Comment by valentinbreiz** [comment]
Done !

**Comment by jp2masa** [comment]
Can you do this small change too, please? https://github.com/CosmosOS/Cosmos/pull/734/files#r132822920

**Comment by valentinbreiz** [comment]
Done! xD

**Pull request merged**